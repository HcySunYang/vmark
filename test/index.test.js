const vmark = require('../')

test('main', () => {
  const markdown = `
---
components:
  counter: ./counter.vue
---
# hello

<counter :start="0" />
`.trim()

  expect(vmark(markdown)).toMatchSnapshot()
})
