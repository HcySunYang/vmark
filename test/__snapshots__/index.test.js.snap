// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
Object {
  "attributes": Object {
    "components": Object {
      "counter": "./counter.vue",
    },
  },
  "component": "import counter from './counter.vue'

export default {
  render: function render() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c(
      'div',
      { staticClass: 'vmark' },
      [
        _c('h1', { attrs: { id: 'hello' } }, [_vm._v('hello')]),
        _vm._v(' '),
        _c('counter', { attrs: { start: 0 } })
      ],
      1
    )
  },
  components: {
    counter: counter
  }
}
",
}
`;

exports[`data 1`] = `
Object {
  "attributes": Object {
    "data": Object {
      "foo": 1,
    },
  },
  "component": "export default {
  data: function data() {
    return {
      foo: 1
    }
  },

  render: function render() {
    var _vm = this
    var _h = _vm.$createElement
    var _c = _vm._self._c || _h
    return _c('div', { staticClass: 'vmark' }, [
      _c('p', [_vm._v(_vm._s(_vm.foo))])
    ])
  }
}
",
}
`;
